src/main/java/insat/company/platform/web/rest/ClubResource.java

```diff
@@ -127,4 +137,59 @@ public ClubResource(ClubService clubService) {
         return clubService.search(query);
     }
 
+    /**
+     * GET  /join/clubs/:id : send joinClubRequest by  club "id"
+     *
+     * @param id the id of the club  to join
+     * @return the ResponseEntity with status 200 (OK), or with status 404 (Not Found) or with status 401 (unauthorized)
+     */
+    @GetMapping("/join/clubs/{id}")
+    @Timed
+    public ResponseEntity<?> joinClub(@PathVariable Long id) throws URISyntaxException {
+        log.debug("REST request to create a joinClubRequest to join a shouldReturnOkAndCreateAJoinClubRequest\nclub   : {}", id);
+
+        if (!SecurityUtils.isAuthenticated()) {
+            log.error("User should be logged in");
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+
+        String userLogin = SecurityUtils.getCurrentUserLogin().get();
+
+        return clubService.findOne(id).map(club -> {
+            User currentUser = userService.getUserWithAuthoritiesByLogin(userLogin).get();
+            if(club.hasMember(currentUser))
+                return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+            JoinClubRequest joinClubRequest= clubService.sendClubJoinRequest(club, currentUser);
+            if(joinClubRequest.getId() >= 0L)
```

Comment:
I don't get why are you doing this

Reply:
by doing that we prevent a member from saving a new joinrequest and the >= 0L is because we want to check if a joinrequest was sent (if id==-1L then none was created)